README.md

```diff
@@ -1,66 +1,145 @@
 
-* Processes new/updated/deleted `VolumeSnapshotContents`: The snapshotter only processes `VolumeSnapshotContent` in which the CSI driver specified in the spec matches the controller's driver name.
-  * If the `VolumeSnapshotRef` is set to nil, skip this content since it is not bound to any snapshot object.
-  * Otherwise, the controller verifies whether the content object is correctly bound to a snapshot object. In case the `VolumeSnapshotRef.UID` is set but it does not match its snapshot object or snapshot no long exists, the content object and its associated snapshot will be deleted.
+The CSI external-snapshotter sidecar talks to CSI over socket (/run/csi/socket by default, configurable by -csi-address).
 
 ## Usage
 
-### Running on command line
+The Volume Snapshot feature now depends on a new, volume snapshot controller in addition to the volume snapshot CRDs. Both the volume snapshot controller and the CRDs are independent of any CSI driver. Regardless of the number CSI drivers deployed on the cluster, there must be only one instance of the volume snapshot controller running and one set of volume snapshot CRDs installed per cluster.
 
-For debugging, it is possible to run snapshotter on command line. For example,
+Therefore, it is strongly recommended that Kubernetes distributors bundle and deploy the controller and CRDs as part of their Kubernetes cluster management process (independent of any CSI Driver).
 
-```bash
-csi-snapshotter -kubeconfig ~/.kube/config -v 5 -csi-address /run/csi/socket
-```
+If your cluster does not come pre-installed with the correct components, you may manually install these components by executing the following steps.
 
-### Running in a statefulset
+Install Snapshot Beta CRDs:
+* kubectl create -f config/crd
+* https://github.com/kubernetes-csi/external-snapshotter/tree/master/config/crd
+* Do this once per cluster
 
-It is necessary to create a new service account and give it enough privileges to run the snapshotter. We provide .yaml files that deploy for use together with the hostpath example driver. A real production deployment must customize them:
+Install Common Snapshot Controller:
+* kubectl create -f deploy/kubernetes/snapshot-controller
+* https://github.com/kubernetes-csi/external-snapshotter/tree/master/deploy/kubernetes/snapshot-controller
+* Do this once per cluster
 
-```bash
-for i in $(find deploy/kubernetes -name '*.yaml'); do kubectl create -f $i; done
-```
+Install CSI Driver:
+* Follow instructions provided by your CSI Driver vendor.
+* Here is an example to install the sample hostpath CSI driver
+  * kubectl create -f deploy/kubernetes/csi-snapshotter
+  * https://github.com/kubernetes-csi/external-snapshotter/tree/master/deploy/kubernetes/csi-snapshotter
 
-### Running with Leader Election
+### Important changes in the snapshot beta APIs
```

Comment:
clarify from alpha1v1-> beta1v1

Reply:
should this go into changelog instead?

Reply:
Sure, I'll add this to the changelog, but I still want to highlight some of these in this doc.  Let me rephrase a little bit.